File: {{cookiecutter.project_slug}}/src/{{cookiecutter.project_file_name}}.py
#!/usr/bin/env python3

"""{{cookiecutter.project_short_description}}

Docstring long description
"""
import configparser
import logging, logging.config
import os
from pathlib import Path
from pprint import pprint
import sys
import time

import numpy as np
import pandas as pd
import requests
import yaml

import globals

__author__ = "{{cookiecutter.full_name}}"
__created__ = "{{cookiecutter.date}}"
__copyright__ = "Copyright 2019, CHE Proximity"
__credits__ = ["{{cookiecutter.full_name}}"]

__version__ = "{{cookiecutter.version}}"
__maintainer__ = "{{cookiecutter.full_name}}"
__email__ = "{{cookiecutter.email}}"
__status__ = "Development" # "Development", "Prototype", or "Production"


def setup_logging(default_path, default_level=logging.WARNING):
    """Setup logging configuration

    """
    path = default_path
    if path.exists():
        with open(path, 'rt') as f:
            config = yaml.safe_load(f.read())
        logging.config.dictConfig(config)
    else:
        logging.basicConfig(level=default_level)
        logging.warning('logging.yaml not imported')


def main():
    pass


if __name__ == "__main__":
    script_start = time.time()
    try:
        setup_logging(globals.BASE_DIR / 'logging_config.yaml', logging.DEBUG) # todo fix logging
        logging.info("__main__ start")

        main()

        logging.info("__main__ end successfully")
    except Exception as e:
        logging.error("__main__ end with error")
        raise
    finally:
        print(f"the script took {time.time() - script_start}s to complete")

File: {{cookiecutter.project_slug}}/src/{{cookiecutter.project_file_name}}.R
#{{cookiecutter.project_short_description}}
#
# {{cookiecutter.project_file_name}}
# Docstring long (multiline) description
#
#
# author = {{cookiecutter.full_name}}
# created = {{cookiecutter.date}}
# copyright = "Copyright 2019, CHE Proximity"
# credits = ["{{cookiecutter.full_name}}", ]# List of people who worked on the project
#
# version = {{cookiecutter.version}}
# maintainer = {{cookiecutter.full_name}}
# email = {{cookiecutter.email}}
# status = Development # Development, Prototype, or Production

pacman::p_load(configr,
               odbc,
               aws.s3,
               here,
               update= FALSE ,install=TRUE)
​
#clear all variables
​
rm(list = ls(all.names = TRUE))
# store setiing file paths and read files
​
p_user_setting_loc = file.path(gsub("\\\\","/",as.character(Sys.getenv(x=c('USERPROFILE'),names=TRUE))),"user_settings.config",fsep = "/")
user_config = read.config(p_user_setting_loc, file.type = 'ini')
​
p_project_setting_loc = file.path(here(),"project_settings.config",fsep="/")
project_config = read.config(p_project_setting_loc, file.type = 'ini')
​
# aws_chepdata_con = odbc::dbConnect(odbc(),
#                                    Driver = user_config$AWS_MSSQL_CHEPDATADB$DRIVER,
#                                    Server = user_config$AWS_MSSQL_CHEPDATADB$ENDPOINT,
#                                    Database = user_config$AWS_MSSQL_CHEPDATADB$DATABASE,
#                                    UID = user_config$AWS_MSSQL_CHEPDATADB$USERNAME,
#                                    PWD = user_config$AWS_MSSQL_CHEPDATADB$PASSWORD,
#                                    Port = as.numeric(user_config$AWS_MSSQL_CHEPDATADB$PORT))
# 
# odbc::dbDisconnect(aws_chepdata_con)
#
# sam_ce <- aws.s3::s3read_using(FUN = read.csv,
#                                bucket = <bucket name>,
#                                object = <>)

#  aws.s3::s3write_using(FUN = write.csv,
# 							x=df,
#                           bucket = <bucket name>,
#                           object = <>)



